src/Microsoft.NetCore.Analyzers/Core/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -246,5 +257,34 @@ private static ParameterInfo GetParameterInfo(INamedTypeSymbol type, bool isArra
         {
             return ParameterInfo.GetParameterInfo(type, isArray, arrayRank, isParams);
         }
+
+        private static bool HasCorrectTryMethodParameters(IMethodSymbol candidateMethod, IMethodSymbol referenceMethod)
+        {
+            if (candidateMethod.Parameters.Length <= referenceMethod.Parameters.Length)
+            {
+                return false;
+            }
+
+            var remainingCandidateMethodParameters = candidateMethod.Parameters.ToList();
+            for (int i = 0; i < referenceMethod.Parameters.Length; i++)
+            {
+                for (int j = 0; j < remainingCandidateMethodParameters.Count; j++)
```

Comment:
I think the implementation here needs to be hardened. We need to ensure that `candidateMethod.Parameters.Length == referenceMethod.Parameters.Length + 1`, and the extra parameter of candidateMethod has `IFormatProvider` type and parameter types of both methods, excluding that parameter from candidateMethod, exactly match both in their indices and type.

Reply:
There is a problem with this implementation for date and times as they have overload with more than 1 parameter being valid.